salsa20/src/lib.rs

```diff
@@ -198,13 +221,12 @@ impl<R: Unsigned> StreamCipherSeekCore for SalsaCore<R> {
 
     #[inline(always)]
     fn get_block_pos(&self) -> u64 {
-        (self.state[8] as u64) + ((self.state[9] as u64) << 32)
+        self.state[8] as u64
     }
 
     #[inline(always)]
     fn set_block_pos(&mut self, pos: u64) {
-        self.state[8] = (pos & 0xffff_ffff) as u32;
-        self.state[9] = ((pos >> 32) & 0xffff_ffff) as u32;
+        self.state[8] = pos as u32;
     }
```

Comment:
This looks like you're switching from a 64-bit to a 32-bit counter?

Reply:
Thank you for pointing me to that. It seems i changed the counter a while back because the `xsalsa_seek` test was not passing but i never went back to it to fix.

The problem is that when we are in sse2 mode the state is shuffled so it needs to get and set the counter in the right positions.

This was fixed in https://github.com/RustCrypto/stream-ciphers/pull/348/commits/98bf2043d863275a14684c9327e41f9bab7ef763